alf/algorithms/algorithm.py

```diff
@@ -154,25 +290,51 @@ def train_complete(self,
                 It is batched from each `info` returned bt `train_step()`
             valid_masks (tf.Tensor): masks indicating which samples are valid.
                 shape=(T, B), dtype=tf.float32
+            weight (float): weight for this batch. Loss will be multiplied with
+                this weight before calculating gradient
         Returns:
             loss_info (LossInfo): loss information
             grads_and_vars (list[tuple]): list of gradient and variable tuples
         """
-
         with tape:
             loss_info = self.calc_loss(training_info)
             if valid_masks is not None:
                 loss_info = tf.nest.map_structure(
-                    lambda l: tf.reduce_mean(l * valid_masks), loss_info)
+                    lambda l: tf.reduce_mean(l * valid_masks)
```

Comment:
when is the shape not equal to 2 ,  and why is the `else` branch doing the same thing

Reply:
When it's not 2 it's expected to be a scalar.